flac.md

```diff
@@ -654,6 +654,57 @@ The reason for this limit is to ensure that decoders can use 32-bit integers whe
+
+## Matroska mapping
+
+The Matroska container format is defined in [@!I-D.ietf-cellar-matroska]. The codec ID (EBML path `\Segment\Tracks\TrackEntry\CodecID`) assigned to signal tracks carrying FLAC data is `A_FLAC` in ASCII. All FLAC data before the first audio frame (i.e. the `fLaC` ASCII signature and all metadata blocks) is stored as CodecPrivate data (EBML path `\Segment\Tracks\TrackEntry\CodecPrivate`).
+
+Each FLAC frame (including all of its subframes) is treated as a single frame in the Matroska context.
+
+In case an audio stream is encoded where audio parameters (sample rate, number of channels or bit depth) change at some point in the stream, this should be dealt with a by finishing the current Matroska segment and starting a new one with the new parameters.
+
+## ISO Base Media File Format (MP4) mapping
+
+The full encapsulation definition of FLAC audio in MP4 files was deemed too extensive to include in this document. A definition document can be found at https://github.com/xiph/flac/blob/master/doc/isoflac.txt This document is summarized here.
+
+The sample entry code is 'fLaC'. No object type indication is registered. The channelcount and samplesize fields in the sample entry follow from the values as found in the FLAC stream. The samplerate field can be different, because FLAC can carry audio with much higher samplerates than can be coded for in the sample entry. When possible, the samplerate field should contain the sample rate as found in the FLAC stream, shifted left by 16 bits to get the 16.16 fixed point representation of the samplerate field. When the FLAC stream contains a sample rate higher than can be coded, the samplerate field contains the greatest expressible regular division of the sample rate, e.g. 48000 for sample rates of 96kHz and 192kHz or 44100 for a sample rate of 88200Hz. When the FLAC stream contain audio with an unusual samplerate that has no regular division, the maximum value of 65535.0 Hz is used. As FLAC streams with a high sample rate are common, a parser or decoder MUST read the value from the FLAC streaminfo metadata block or a frame header to determine the actual sample rate. The sample entry contains one 'FLAC specific box' with code 'dfLa'.
+
+The FLAC specific box has version number 0 and contains the metadata blocks of the FLAC stream. Contrary to the CodecPrivate data in the Matroska mapping, this excludes the `fLaC` ASCII signature.
+
+In case an audio stream is encoded where audio parameters (sample rate, number of channels or bit depth) change at some point in the stream, this should be dealt with **TODO: Someone with knowledge of this should chime in. This is NOT handled in the mapping document**
+
+Each FLAC frame is a single sample in the context of MP4 files.
```

Comment:
```suggestion
A MP4 sample SHOULD contain one FLAC frame.
```

IIRC this form is usually seen in MP4 related docs.

Reply:
Wouldn't this result in the question: what are the exceptions to this? If I remember correctly from the recent Matroska discussion, a SHOULD without documented exceptions is a MUST. What about simply

```suggestion
A MP4 sample contains one single FLAC frame.
```

Reply:
Technically MP4 does not limit the count of frames per MP4 samples, but all documents limiting the possibilities of MP4 indicate this limitation, as if it is something we should be afraid about. But I don't see any problem with more than 1 frame in a MP4 sample, reason I have put "SHOULD", but I am fine to remove it.

Reply:
One problem with FLAC is that the length of a frame is not explicitly stored anywhere. I can imagine that with PCM audio, where a frame is generally only a few bytes and of fixed length, storing several frames in a sample makes sense, but I think it needlessly complicates demuxing.

Also, this bit of text is in my view only meant to be a summary of the full document, which states the following
```
            A FLAC sample is exactly one FLAC audio FRAME (as defined
            in the FLAC[3] file specification) belonging to a FLAC
            bitstreams.
```

Reply:
I am usually reluctant to limit a container format, but to be honest 1 frame per sample is a classic rule, so fine for me.